test/libdnf/rpm/test_solv_query.cpp

```diff
@@ -30,402 +31,423 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+    CPPUNIT_ASSERT_EQUAL(expected, actual);
+
+    // ---
+
+    // packages with Name that contain "kG-l" - case insensitive match
     libdnf::rpm::SolvQuery query8(sack);
-    std::vector<std::string> names_icontains{"qRli"};
-    query8.ifilter_name(libdnf::sack::QueryCmp::ICONTAINS, names_icontains);
-    CPPUNIT_ASSERT_EQUAL(4lu, query8.size());
-    for (auto pkg : query8) {
-        CPPUNIT_ASSERT(nevras_contains.find(pkg.get_nevra()) != nevras_contains.end());
-    }
-    query8.ifilter_name(libdnf::sack::QueryCmp::CONTAINS, names_icontains);
-    CPPUNIT_ASSERT_EQUAL(0lu, query8.size());
+    query8.ifilter_name(libdnf::sack::QueryCmp::ICONTAINS, {"kG-l"});
 
-    // Test unsupported cmp type
-    CPPUNIT_ASSERT_THROW(query8.ifilter_name(libdnf::sack::QueryCmp::GT, names_icontains);
-                         , libdnf::rpm::SolvQuery::NotSupportedCmpType);
+    expected = {"pkg-libs-0:1.2-3.x86_64", "pkg-libs-1:1.2-4.x86_64", "pkg-libs-1:1.3-4.x86_64"};
+    actual = to_vector(query7);
+    CPPUNIT_ASSERT_EQUAL(expected, actual);
 
-    // Test QueryCmp::EQ with two elements
+    // ---
+
+    // unsupported comparison type (operator)
+    CPPUNIT_ASSERT_THROW(
+        query8.ifilter_name(libdnf::sack::QueryCmp::GT, {"pkg"}),
+        libdnf::rpm::SolvQuery::NotSupportedCmpType
+    );
+
+    // ---
+
+    // packages with Name "pkg" or "pkg-libs" - two patterns matched in one expression
     libdnf::rpm::SolvQuery query9(sack);
-    std::vector<std::string> names3{"CQRlib", "nodejs"};
-    query9.ifilter_name(libdnf::sack::QueryCmp::EQ, names3);
-    CPPUNIT_ASSERT_EQUAL(4lu, query9.size());
-    for (auto pkg : query9) {
-        CPPUNIT_ASSERT(full_nevras.find(pkg.get_full_nevra()) != full_nevras.end());
-    }
+    query9.ifilter_name(libdnf::sack::QueryCmp::EQ, {"pkg", "pkg-libs"});
+
+    expected = {"pkg-0:1.2-3.src", "pkg-0:1.2-3.x86_64", "pkg-libs-0:1.2-3.x86_64", "pkg-libs-1:1.2-4.x86_64", "pkg-libs-1:1.3-4.x86_64"};
+    actual = to_vector(query9);
+    CPPUNIT_ASSERT_EQUAL(expected, actual);
 }
 
-void RpmSolvQueryTest::test_ifilter_nevra() {
-    std::set<std::string> nevras{"CQRlib-0:1.1.1-4.fc29.src", "CQRlib-0:1.1.1-4.fc29.x86_64"};
 
+void RpmSolvQueryTest::test_ifilter_nevra() {
     {
         // Test QueryCmp::EQ - argument without 0 epoch - two elements
         libdnf::rpm::SolvQuery query(sack);
-        std::vector<std::string> nevras_without_0_epoch{"CQRlib-1.1.1-4.fc29.src", "CQRlib-1.1.1-4.fc29.x86_64"};
-        query.ifilter_nevra(libdnf::sack::QueryCmp::EQ, nevras_without_0_epoch);
-        CPPUNIT_ASSERT_EQUAL(2lu, query.size());
-        for (auto pkg : query) {
-            CPPUNIT_ASSERT(nevras.find(pkg.get_full_nevra()) != nevras.end());
-        }
+        query.ifilter_nevra(libdnf::sack::QueryCmp::EQ, {"pkg-1.2-3.src", "pkg-1.2-3.x86_64"});
+        std::vector<std::string> expected = {"pkg-0:1.2-3.src", "pkg-0:1.2-3.x86_64"};
+        std::vector<std::string> actual = to_vector(query);
```

Comment:
I'd prefer to use `to_vector(query)` directly in the assert below, no need to use an intermediate variable for this one. It decreases verbosity and (subjectively) readability. The line count adds up with all the repetition below.

Reply:
Changed. The suggested change makes sense, the code is shorter and better readable.